packages/runtime-core/src/componentOptions.ts

```diff
@@ -138,7 +138,7 @@ export interface ComponentOptionsBase<
   // Luckily `render()` doesn't need any arguments nor does it care about return
   // type.
   render?: Function
-  components?: Record<string, Component>
+  components?: Record<string, Component | { __isTeleport: true } | { __isSuspense: true } | { __isKeepAlive: true }>
```

Comment:
Same problem 😅

Reply:
I think components from user's land (including  RouterView or RouterLink  ) shall server as `Component` type,  only built-in components Teleport, Suspense and KeepAlive need special treatment